[docs] Migrate sorting (#233)
mui-org/material-ui-x: docs/src/pages/components/data-grid/rows/rows.md

## Row sorting

### Basic sorting

> Single column sorting can be triggered with one click on column header.

Reviewer: ```suggestion
Single column sorting can be triggered with by clicking a column header.
```

> You can repeat this action to change the sorting direction.

Reviewer: ```suggestion
Repeat this action to change the sorting direction.
```

> You can pre-configure a sorted column using the `sortModel` prop of the `ColDef` interface as below:

Reviewer: ```suggestion
A sorted column can be  can pre-configured using the `sortModel` prop of the `ColDef` interface:
```

```ts
const sortModel = [
  {
    field: 'commodity',
    sort: 'asc' as SortDirection,
  },
];
```

{{"demo": "pages/components/data-grid/rows/BasicSortingGrid.js"}}

### Custom comparator

The Grid handles sorting and applies different comparators in columns according to their types.
The component handles sorting natively for the following types:

- string
- number
- date

> To extend or modify this behavior in a specific column, you can pass a custom comparator, and override the `sortComparator` prop of the `ColDef` interface.

Reviewer: ```suggestion
To extend or modify this behavior in a specific column, you can pass in a custom comparator, and override the `sortComparator` prop of the `ColDef` interface.
```

> In the example below, the `username` column combines `name` and `age` but it is sorted by `age` using a custom comparator:

Reviewer: ```suggestion
In the example below, the `username` column combines `name` and `age`, but it is sorted by `age` using a custom comparator:
```

{{"demo": "pages/components/data-grid/rows/ComparatorSortingGrid.js"}}

> ### Sorting order

Reviewer: ```suggestion
### Sort order
```

> By default, the sorting order loops between these tree different modes:

Reviewer: ```suggestion
By default, the sort order cycles between these three different modes:
```

```suggestion
By default, the sort order rotates between these three different modes:
```

```jsx
const sortingOrder = ['asc', 'desc', null];
```

In practice, when you click a column that is not sorted, it will sort ascending (`asc`).

> The next click will make it sort descending (`desc`). Another click will remove the sort (`null`), using the orginal order the data was provided in.

Reviewer: ```suggestion
The next click will make it sort descending (`desc`). Another click will remove the sort (`null`), reverting to the order that the data was provided in.
```

This behavior can be overwritten by setting the `sortingOrder` prop.

> In the example below columns are only sortable in descending then ascending order.

Reviewer: ```suggestion
In the example below columns are only sortable in descending or ascending order.
```

{{"demo": "pages/components/data-grid/rows/OrderSortingGrid.js"}}

### Disable sorting

> By default, columns are all sortable and it can be revoked using the sortable prop of the `ColDef` interface as below.

Reviewer: ```suggestion
By default, all columns are sortable. This can be revoked using the sortable prop of the `ColDef` interface:
```

```tsx
const columns: ColDef = [{ field: 'name', sortable: false }];
```

{{"demo": "pages/components/data-grid/rows/DisableSortingGrid.js"}}

### Server-side sorting

> By default, sorting works on the client-side.

Reviewer: ```suggestion
By default, sorting works client-side.
```

To switch it to server-side, set `sortingMode="server"`.

> Then, you need to handle the `onSortModelChange` callback, sort the rows on the server-side and update the `rows` prop with the new sorted rows.

Reviewer: ```suggestion
Then you need to handle the `onSortModelChange` callback, sort the rows on the server-side, and update the `rows` prop with the newly sorted rows.
```

{{"demo": "pages/components/data-grid/rows/ServerSortingGrid.js"}}

### apiRef

> The Grid exposes a set of methods that will let you achieve all the above features using the imperative apiRef.

Reviewer: ```suggestion
The Grid exposes a set of methods that supports all of the above features using the imperative apiRef.
```

> > ⚠️ Only use this API when you have no alternatives. Always start from the declarative API the Grid exposes.

Reviewer: ```suggestion
> ⚠️ Only use this API when you have no alternative. Always start from the declarative API that the Grid exposes.
```
